### PR TITLE
ui: refactor model related functions

### PR DESCRIPTION
**Pull request by deanlee**
- read model data with c++-capnp patterns
- delete structs in modeldata.h designed for c-capnp
- include modeldata.h in modeld to remove duplicated defines

### CODE REVIEW

File: selfdrive/ui/ui.hpp [hunk 1]
```diff
@@ -126,6 +124,10 @@ typedef struct UIScene {
   cereal::ControlsState::Reader controls_state;
   cereal::DriverState::Reader driver_state;
   cereal::DMonitoringState::Reader dmonitoring_state;
+  cereal::ModelData::Reader model;
+  float left_path_points[MODEL_PATH_DISTANCE];
```
Comment (pd0wm) [inline selfdrive/ui/ui.hpp:128]: left_lane_points?

**Comment by pd0wm** [comment]
Can you fix the conflicts?

**Comment by deanlee** [comment]
fixed

**Comment by pd0wm** [comment]
Thanks!

This PR introduces a change where the path is only drawn up to path.validLen. Waiting for confirmation from Harald that this value is good to use in the ui.

**Pull request merged**